buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/electordesinterface.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+from typing import Union
+
+import numpy as np
+from neuroconv import BaseDataInterface
+from neuroconv.datainterfaces.ecephys.baserecordingextractorinterface import (
+    BaseRecordingExtractorInterface,
+)
+from neuroconv.tools.spikeinterface.spikeinterface import add_electrodes
+from pymatreader import read_mat
+from pynwb import NWBFile
+from spikeinterface.core.numpyextractors import NumpyRecording
+
+
+class HuszarElectrodeInterface(BaseDataInterface):
+    def __init__(self, folder_path: Union[str, Path], verbose: bool = True):
+        folder_path = Path(folder_path)
+        self.chan_map_file_path = folder_path / "chanMap.mat"
+        assert self.chan_map_file_path.exists(), f"chanMap.mat file not found in {folder_path}"
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
+        mat_file = read_mat(self.chan_map_file_path)
+        channel_groups = mat_file["connected"]
+        channel_group_names = [f"Group{group_index + 1}" for group_index in channel_groups]
+
+        channel_indices = mat_file["chanMap0ind"]
+
+        # Follow Neuroscope convention
+        channel_ids = [str(channel_indices[i]) for i in channel_indices]
+        channel_name = [
+            f"ch{channel_index}grp{channel_group}"
+            for channel_index, channel_group in zip(channel_indices, channel_groups)
+        ]
+
+        x_coords = mat_file["xcoords"]
+        y_coords = mat_file["ycoords"]
+        locations = np.array((x_coords, y_coords)).T.astype("float32")
+
+        num_channels = len(channel_indices)
+        traces_list = [np.ones(shape=(1, num_channels))]
+        sampling_frequency = 30_000.0
+        recording = NumpyRecording(
+            traces_list=traces_list, sampling_frequency=sampling_frequency, channel_ids=channel_ids
+        )
+        recording.set_channel_locations(channel_ids=channel_ids, locations=locations)
+
+        recording.set_property(key="channel_name", values=channel_name)
+
+        # This should transform to microvolts which is the standard in spikeinterface
+        recording.set_channel_gains(channel_ids=channel_ids, gains=np.ones(num_channels) * 0.195)
+        recording.set_channel_offsets(channel_ids=channel_ids, offsets=np.ones(num_channels) * 32_768)
```

Comment:
Whoa whoa, do offsets get set to the real extractor? Or to a corresponding intan reading of the info.rhd?

Thing is, this is signed data - usually that offset is only for unsigned intan data

I could have sworn that when using SI to read this stuff the offset is only set if its unsigned

Reply:
Thanks for pointing that out. I am losing a multiplicative factor. Look at how the oficial implementation for reading intan that intan systems provides looks:

https://github.com/Intan-Technologies/load-rhd-notebook-python/blob/9026c8c0ac0693ed4896c2f50f55a3f57237bb68/importrhdutilities.py#L579

But I think they use the offset there is because they initially define the data as unsigned:
https://github.com/Intan-Technologies/load-rhd-notebook-python/blob/9026c8c0ac0693ed4896c2f50f55a3f57237bb68/importrhdutilities.py#L511

Which sounds like what are you saying as Neuroscope already loads the data signed and then has the offset 0.
https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/neuroscoperawio.py#L111

I have become very suspicious of that Neuroscope reader but if indeed we are going to read the data using it, we should probably set the offest to 0.

The other readers for intan have the offset though:

pyintan has it though:
https://github.com/alejoe91/pyintan/blob/92ee10acad13bd2b5207a158a2e38ce222be716a/pyintan/intan.py#LL108C19-L108C19

And the reader for intan in neo:
https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/intanrawio.py#L491

Reply:
I know the base readers underlying pyintan and neo always set it, my contention was that something in the SI loader would make the decision whether or not to set that to the extractor, somewhere around https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/extractors/neoextractors/neobaseextractor.py#L156

At least the last time I used the intan reader on data, that's how it worked (I've ran before on both signed and unsigned intan data) - point being, the offset was only needed when the data was unsigned

Anyway, yes for Neuroscope it should soundly be zero, since it was acquired as signed intan (which also should not use offsets, which if SI currently does is, I suspect, a bug)

Reply:
I commented out the line because we are going to use Neuroscope. I also tested Neuroscope for the valero data and the offset is zero.

Beside that, 
Are you saying that using SI to read intan data should set the offsets to zero? Not clear to me why and it is not the case if I read Valero data with Intan:

![image](https://github.com/catalystneuro/buzsaki-lab-to-nwb/assets/6429509/d7728827-3cc6-4333-b61f-8c44a364d1a6)

I don't really understand what do you mean by "acquiring with signed intan", can you elaborate?

Reply:
Got you, yeah, I am confused about this. Pyintan has this which indicates that there are some versions where the data is typed:

https://github.com/NeuralEnsemble/python-neo/blob/f608309c5ce031ecd905349c140b07a3dafb057d/neo/rawio/intanrawio.py#L473-L491

But I can't find any references to it in the oficial repo provided by the company...